apps/builder/app/builder/features/settings-panel/props-section/animation/subject-select.tsx

```diff
@@ -107,51 +99,69 @@ export const SubjectSelect = ({
         const selector =
           subjects.find((s) => s.value === subject)?.selector ?? undefined;
         $hoveredInstanceSelector.set(selector);
+
+        if (subject === undefined) {
+          onChange(undefined, true);
+          return;
+        }
+
+        const subjectItem = subjects.find((s) => s.value === subject);
+        if (subjectItem === undefined) {
+          onChange(undefined, true);
+          return;
+        }
+        const newValue = {
+          ...value,
+          subject: subject === "self" ? undefined : subject,
+        };
+
+        onChange(newValue, true);
       }}
       onChange={(subject) => {
         const newValue = {
           ...value,
           subject: subject === "self" ? undefined : subject,
         };
-        const parsedValue = animationActionSchema.safeParse(newValue);
-
-        if (parsedValue.success) {
-          const subjectItem = subjects.find((s) => s.value === subject);
-
-          if (subjectItem === undefined) {
-            toast.error(`Subject "${newValue.subject}" not found`);
-            return;
-          }
-
-          if (
-            subjectItem.isTimelineExists === false &&
-            newValue.subject !== undefined
-          ) {
-            serverSyncStore.createTransaction(
-              [$breakpoints, $styles, $styleSources, $styleSourceSelections],
-              (breakpoints, styles, styleSources, styleSourceSelections) => {
-                if (newValue.subject === undefined) {
-                  return;
-                }
-
-                setListedCssProperty(
-                  breakpoints,
-                  styleSources,
-                  styleSourceSelections,
-                  styles
-                )(subjectItem.instanceId, "view-timeline-name", {
-                  type: "unparsed",
-                  value: newValue.subject,
-                });
+
+        const subjectItem = subjects.find((s) => s.value === subject);
```

Comment:
opens the door to use single char variables everywhere, I know its handy but its not worth it
```suggestion
        const subjectItem = subjects.find(({value}) => value === subject);
```